### ISSUE TITLE
All samples have inkling that are missing rewards or goal functions

### ISSUE THREAD

**Issue by journeyman-msft** [microsoft/bonsai-gym#1]

**Comment by akzaidi** [comment]
The gym terminals and rewards were added to `state` in each of `GymSimulator3`'s methods for `get_state` and `episode_finish`, so now you can simply create reward and terminal functions that return them:

```
function Reward(State: SimState) {

    return State._gym_reward
}

function Terminal(State: SimState) {

    return State._gym_terminal
}

graph (input: GameState): Action {
    concept GoalPosition(input): Action {
        curriculum {
            source FrozenlakeSimulator
            reward Reward
            terminal Terminal
        }
    }
    output GoalPosition
}

```

Just remember to filter out those from your MDP otherwise your brain will just learn an identity map!

```

type GameState {
    current_pos: Number.Int8<0 .. 15>,
}

type SimState {
    current_pos: Number.Int8<0 .. 15>,
    _gym_reward: number,
    _gym_terminal: Number.Bool
}
```

We can update all the inklings with this?

**Comment by journeyman-msft** [comment]
Let's keep this issue until we get all the Inklings up to date I suppose.

**Comment by akzaidi** [comment]
Should be resolved by #4

**Comment by nmann4** [comment]
One more PR to finish the SimState / GameState update on the samples #5 .
